backend/src/main/java/ch/puzzle/okr/service/business/QuarterBusinessService.java

```diff
@@ -72,12 +84,37 @@ public YearMonth getCurrentYearMonth() {
         return YearMonth.now();
     }
 
+    public Map<Integer, Integer> generateQuarters() {
+        Map<Integer, Integer> quarters = new HashMap<>();
+        int quarterIndex = quarterStart - 1;
+        for (int i = 0; i < 12; i++) {
+            // Get the index of the current month based on quarterStart
+            int monthIndex = (i + quarterIndex) % 12;
+            // Get the quarter (three months), +1 since we don't want to start with 0
+            int quarter = (i / 3) + 1;
+            // Return the real month
+            quarters.put(monthIndex + 1, quarter);
+        }
+
+        return quarters;
+    }
+
     @Scheduled(cron = "0 59 23 L * ?") // Cron expression for 23:59:00 on the last day of every month
     public void scheduledGenerationQuarters() {
-        YearMonth yearMonth = getCurrentYearMonth();
-        if (yearMonth.getMonthValue() % 3 == 0) {
+        Map<Integer, Integer> quarters = generateQuarters();
+        YearMonth currentYearMonth = getCurrentYearMonth();
+        YearMonth yearMonthToGenerate = currentYearMonth.plusMonths(4);
+
+        int currentQuarter = quarters.get(currentYearMonth.getMonthValue());
+        int nextQuarter = quarters.get(yearMonthToGenerate.getMonthValue());
+
+        // If we are in the last month of a quarter, generate the next quarter
+        // If the quarter 4 months in the future and the current are exactly 2 apart, this is the case
+        // 1 -> 3, 2 -> 4, 3 -> 1, 4 -> 2
+        if (Math.abs(nextQuarter - currentQuarter) == 2) {
```

Comment:
Ich würde diese Logik ``if (Math.abs(nextQuarter - currentQuarter) == 2)`` in eine neue Methode verschieben, z.B. ``areWeInLastMonthOfQuarter()``  und die zweite Hälfte des Kommentars auch in diese neue Methode verschieben.